url647b55da7c_Nikhilkumar13_emuSr_tgz-pJ8-MainJ8/src/backend/Emit.java

```diff
@@ -0,0 +1,416 @@
+        for (int i = 0; i < div.length; i++)
+        {
+            ht.put(div[i].name(), new backend.instructions.DIV());
+
+        }
+
+        LSL[] lsl = backend.InstructionSet.LSL.values();
+        for (int i = 0; i < lsl.length; i++)
+        {
+            ht.put(lsl[i].name(), new backend.instructions.LSL());
+
+        }
+
+        ASR[] asr = backend.InstructionSet.ASR.values();
+        for (int i = 0; i < asr.length; i++)
+        {
+            ht.put(asr[i].name(), new backend.instructions.ASR());
+
+        }
+        NOT[] not = backend.InstructionSet.NOT.values();
+        for (int i = 0; i < not.length; i++)
+        {
+            ht.put(not[i].name(), new backend.instructions.NOT());
+
+        }
+
+        SUB[] sub = backend.InstructionSet.SUB.values();
+        for (int i = 0; i < sub.length; i++)
+        {
+            ht.put(sub[i].name(), new backend.instructions.SUB());
+
+        }
+
+        B[] b = backend.InstructionSet.B.values();
+        for (int i = 0; i < b.length; i++)
+        {
+            ht.put(b[i].name(), new backend.instructions.B());
+
+        }
+
+        BEQ[] beq = backend.InstructionSet.BEQ.values();
+        for (int i = 0; i < beq.length; i++)
+        {
+            ht.put(beq[i].name(), new backend.instructions.BEQ());
+
+        }
+        BGT[] bgt = backend.InstructionSet.BGT.values();
+        for (int i = 0; i < beq.length; i++)
+        {
+            ht.put(bgt[i].name(), new backend.instructions.BGT());
+
+        }
+
+        LD[] ld = backend.InstructionSet.LD.values();
+        for (int i = 0; i < ld.length; i++)
+        {
+            ht.put(ld[i].name(), new backend.instructions.LD());
+
+        }
+
+        ST[] st = backend.InstructionSet.ST.values();
+        for (int i = 0; i < st.length; i++)
+        {
+            ht.put(st[i].name(), new backend.instructions.ST());
+
+        }
+        OR[] or = backend.InstructionSet.OR.values();
+        for (int i = 0; i < or.length; i++)
+        {
+            ht.put(or[i].name(), new backend.instructions.OR());
+
+        }
+
+        MOD[] mod = backend.InstructionSet.MOD.values();
+        for (int i = 0; i < mod.length; i++)
+        {
+            ht.put(mod[i].name(), new backend.instructions.MOD());
+
+        }
+
+        MUL[] mul = backend.InstructionSet.MUL.values();
+        for (int i = 0; i < mul.length; i++)
+        {
+            ht.put(mul[i].name(), new backend.instructions.MUL());
+
+        }
+        NOP[] nop = backend.InstructionSet.NOP.values();
+        ht.put(nop[0].name(), new backend.instructions.NOP());
+        RET[] ret = backend.InstructionSet.RET.values();
+        ht.put(ret[0].name(), new backend.instructions.RET());
+        PRINT[] print = backend.InstructionSet.PRINT.values();
+        for (int i = 0; i < print.length; i++)
+        {
+            ht.put(print[i].name(), new backend.instructions.PRINT());
+        }
+      }
+
+    public static void execute(StringTokenizer st)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 83 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.